### PR TITLE
Tweak TamperMonkey Script | Add Stats + Bug Ticket Identifier -abzdma…

### PR DESCRIPTION
**Pull request by abzdmachinist**

### CODE REVIEW

File: dotorg/dotorg-topic-highlighter.user.js [hunk 1]
```diff
@@ -158,17 +158,68 @@ jQuery(document).ready(function( $ ) {
...
+            `);
+        }
+
+        if (totalToResolveTicket > 0 ) {
+            $( '.bbp-pagination' ).append(`
+            <span class="divider" style="font-size:13px; color:#ddd;">/</span>
+            <span for="tec-topics-to-resolve">
+                To Resolve (<b>`+ totalToResolveTicket +`</b>)
+            </span>
+            `);
```
Comment (bordoni) [inline dotorg/dotorg-topic-highlighter.user.js:202]:
```suggestion
            $( '.bbp-pagination' ).append(`
            <span class="divider" style="font-size:13px; color:#ddd;">/</span>
            <span for="tec-topics-to-resolve">
                To Resolve (<b>${totalToResolveTicket}</b>)
            </span>
            `);
```

File: dotorg/dotorg-topic-highlighter.user.js [hunk 2]
```diff
@@ -158,17 +158,68 @@ jQuery(document).ready(function( $ ) {
...
+
+        var totalOnPageTicket = $( '.topic' ).length;
+        var totalNotLastVoiceTicket = $(".topic:not(.tamper-last-voice)").length;
+        var totalLoggedInTicket = $(".tamper-logged-in-ticket").length;
+        var totalNewTicket = $(".tamper-new-ticket").length;
+        var totalToResolveTicket = $(".tamper-to-resolve").length;
+        var totalFollUpTicket = Math.abs(totalNotLastVoiceTicket - totalNewTicket);
+
+        // Show stats beside the pagination to and bottom
+        $( '.bbp-pagination' ).append( `
```
Comment (bordoni) [inline dotorg/dotorg-topic-highlighter.user.js:181]: Save the `$( '.bbp-pagination' )` into a local variable.

**Review by bordoni — CHANGES_REQUESTED** [review]
All of the spacing on `dotorg-topic-updater.user.js` that was removed needs to be brought back.

File: dotorg/dotorg-topic-highlighter.user.js (hunk 1 above)
Comment (abzdmachinist) [reply dotorg/dotorg-topic-highlighter.user.js]: @bordoni updated.

File: dotorg/dotorg-topic-highlighter.user.js (hunk 2 above)
Comment (abzdmachinist) [reply dotorg/dotorg-topic-highlighter.user.js]: Done adding a local variable. Kindly check @bordoni

File: dotorg/dotorg-topic-highlighter.user.js (hunk 1 above)
Comment (abzdmachinist) [reply dotorg/dotorg-topic-highlighter.user.js]: @bordoni appreciate the comments and suggestions. What format are you using for JS on your editor? I usually have this format for TS, apologies that it was carried over when I saved and forgot to turn it off.

File: dotorg/dotorg-helper.user.js [hunk 3]
```diff
@@ -21,7 +21,7 @@
  * Marks threads yellow that are more than a month old.
  */
 
-(function() {
+ (function() {
```
Comment (Camwyn) [inline dotorg/dotorg-helper.user.js:24]:
```suggestion
(function() {
```

File: dotorg/dotorg-helper.user.js [hunk 4]
```diff
@@ -107,21 +107,36 @@
...
             }
 
+            // Check tickets to resolve
+            var toResolve = x[i].innerHTML.search( /[2-3] (week[s]?)/ );
+            if ( toResolve > 0 ) {
+                x[i].classList.add('tamper-to-resolve');
+                continue;
+            }
+
             // If not resolved, check if tha last voice is a team member
```
Comment (Camwyn) [inline dotorg/dotorg-helper.user.js:123]:
```suggestion
            // If not resolved, check if that last voice is a team member
```

File: dotorg/dotorg-topic-highlighter.user.js [hunk 5]
```diff
@@ -158,17 +158,70 @@ jQuery(document).ready(function( $ ) {
...
+        var totalOnPageTicket = $( '.topic' ).length;
+        var totalNotLastVoiceTicket = $(".topic:not(.tamper-last-voice)").length;
+        var totalLoggedInTicket = $(".tamper-logged-in-ticket").length;
+        var totalNewTicket = $(".tamper-new-ticket").length;
+        var totalToResolveTicket = $(".tamper-to-resolve").length;
+        var totalFollUpTicket = Math.abs(totalNotLastVoiceTicket - totalNewTicket);
+
+		var paginationContainer = $( '.bbp-pagination' );
+
+        // Show stats beside the pagination to and bottom
```
Comment (Camwyn) [inline dotorg/dotorg-topic-highlighter.user.js:182]: I don't understand this comment

File: dotorg/dotorg-last-voice-highlighter.user.js [hunk 6]
```diff
@@ -95,22 +95,29 @@
...
+        var newTickets = x[i].innerHTML.search( 'bbp-topic-reply-count">0' );
+        
+        if ( newTickets > 0 ) {
+            x[i].classList.add('tamper-new-ticket');
+        }
+        
         for( j = 0; j < tecteam.length; j++ ) {
-
+            
             // If not resolved, check if tha last voice is a team member
```
Comment (Camwyn) [inline dotorg/dotorg-last-voice-highlighter.user.js:113]:
```suggestion
            // If not resolved, check if that the last voice is a team member
```

File: dotorg/dotorg-helper.user.js (hunk 3 above)
Comment (abzdmachinist) [reply dotorg/dotorg-helper.user.js:24]: @Camwyn updated.

File: dotorg/dotorg-helper.user.js (hunk 4 above)
Comment (abzdmachinist) [reply dotorg/dotorg-helper.user.js:123]: @Camwyn apologies typo should be "the"

File: dotorg/dotorg-topic-highlighter.user.js (hunk 5 above)
Comment (abzdmachinist) [reply dotorg/dotorg-topic-highlighter.user.js:182]: @Camwyn another typo should be "top and bottom"

File: dotorg/dotorg-last-voice-highlighter.user.js (hunk 6 above)
Comment (abzdmachinist) [reply dotorg/dotorg-last-voice-highlighter.user.js:113]: Should also be "the" @Camwyn

**Comment by abzdmachinist** [comment]
@Camwyn @bordoni, could you guys share coding standards for JS? I could update all of the files here to make it uniform.

**Comment by Camwyn** [comment]
Our coding standards are here: https://the-events-calendar.github.io/products-engineering/docs/code-standards/javascript/

File: dotorg/dotorg-topic-highlighter.user.js [hunk 7]
```diff
@@ -160,15 +160,73 @@ jQuery(document).ready(function( $ ) {
...
+        var totalLoggedInTicket = $(".tamper-logged-in-ticket").length;
+        var totalNewTicket = $(".tamper-new-ticket").length;
+        var totalToResolveTicket = $(".tamper-to-resolve").length;
+        var totalFollUpTicket = Math.abs(totalNotLastVoiceTicket - totalNewTicket);
+
+		var paginationContainer = $( '.bbp-pagination' );
+
+        // Show stats beside the pagination top and bottom
+        paginationContainer.append( `
+            <span for="tec-pagination count" style="margin-left: -5px;">
```
Comment (Camwyn) [inline dotorg/dotorg-topic-highlighter.user.js:189]: Same comment on inline styles.

File: dotorg/dotorg-topic-highlighter.user.js [hunk 8]
```diff
@@ -160,15 +160,73 @@ jQuery(document).ready(function( $ ) {
...
+            <span class="divider" style="font-size:13px; color:#ddd;">/</span>
+            <span for="tec-topics-logged-in">
+                My Tickets (<b>${totalLoggedInTicket}</b>)
+            </span>
+            `);
+        }
+
+        if ( totalToResolveTicket > 0 ) {
+            paginationContainer.append(`
+            <span class="divider" style="font-size:13px; color:#ddd;">/</span>
```
Comment (Camwyn) [inline dotorg/dotorg-topic-highlighter.user.js:205]: Same comment on inline styles.

File: dotorg/dotorg-topic-highlighter.user.js [hunk 9]
```diff
@@ -160,15 +160,73 @@ jQuery(document).ready(function( $ ) {
...
+            <span for="tec-topics-to-resolve">
+				To Resolve (<b>${totalToResolveTicket}</b>)
+            </span>
+            `);
+        }
+
+        if ( totalFollUpTicket > 0 ) {
+            paginationContainer.append(`
+            <span class="divider" style="font-size:13px; color:#ddd;">/</span>
```
Comment (Camwyn) [inline dotorg/dotorg-topic-highlighter.user.js:214]: Same comment on inline styles.

File: dotorg/dotorg-topic-highlighter.user.js [hunk 10]
```diff
@@ -160,15 +160,73 @@ jQuery(document).ready(function( $ ) {
...
+            <span for="tec-topics-follow-up">
+                Follow Up (<b>`+ totalFollUpTicket +`</b>)
+            </span>
+            `);
+        }
+
+        if ( totalNewTicket > 0 ) {
+            paginationContainer.append(`
+            <span class="divider" style="font-size:13px; color:#ddd;">/</span>
```
Comment (Camwyn) [inline dotorg/dotorg-topic-highlighter.user.js:223]: Same comment on inline styles.

File: dotorg/dotorg-topic-updater.user.js [hunk 11]
```diff
@@ -63,5 +63,15 @@
     $( function() {
         obj.init();
     } );
-})( jQuery, {} );
 
+    // Identify bug tickets
+    let bugTicket = $( document ).text().search( /internal bug ticket reference | ticket number | bug report | bug ticket/i );
```
Comment (Camwyn) [inline dotorg/dotorg-topic-updater.user.js:68]: This is very broad - it searches the _entire page contents_. Can we narrow down the selector on this some?

**Review by Camwyn — CHANGES_REQUESTED** [review]
Remove commented out code. One question.

**Pull request closed without merging**